cmake/catkin_package.cmake

```diff
@@ -374,6 +377,15 @@ function(_catkin_package)
       endif()
     endforeach()
 
+    set(PKG_COMPILE_OPTIONS ${${PROJECT_NAME}_COMPILE_OPTIONS})
+    separate_arguments(PKG_COMPILE_OPTIONS)
```

Comment:
> I would expect the arguments to be passed separately and not in a single string

I have seen add_definitions("...") in a couple of projects, so people might pass it as full string because they have found it somewhere in a stackoverflow answer.
IMHO if we need to check for conflicting settings then we should split them.